### PR TITLE
[MLIR] Dump sass

### PR DESCRIPTION
**Pull request by grypp**
This PR dump sass by using nvdiasm

### CODE REVIEW

File: mlir/lib/Target/LLVM/NVVM/Target.cpp [hunk 1]
```diff
@@ -92,6 +93,7 @@ SerializeGPUModuleBase::SerializeGPUModuleBase(
   // If `targetOptions` have an empty toolkitPath use `getCUDAToolkitPath`
   if (toolkitPath.empty())
     toolkitPath = getCUDAToolkitPath();
+  llvm::errs() << toolkitPath;
```
Comment (joker-eph) [inline mlir/lib/Target/LLVM/NVVM/Target.cpp:96]: Debug leftover?

File: mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // RUN: mlir-opt %s \
-// RUN:  | mlir-opt -gpu-lower-to-nvvm-pipeline -debug-only=serialize-to-isa \
+// RUN:  | mlir-opt -gpu-lower-to-nvvm-pipeline -debug-only=serialize-to-isa -debug-only=dump-sass \
```
Comment (joker-eph) [inline mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir:2]: Missing check lines?

**Review by joker-eph — APPROVED** [review]

File: mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir (hunk 2 above)
Comment (grypp) [reply mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir]: It's hard to check SASS as llvm doesn't have control. But I added a check that checks most common instructions.

File: mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // RUN: mlir-opt %s \
-// RUN:  | mlir-opt -gpu-lower-to-nvvm-pipeline -debug-only=serialize-to-isa \
+// RUN: | mlir-opt -gpu-lower-to-nvvm-pipeline -debug-only=serialize-to-isa \
```
Comment (joker-eph) [inline mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir:2]: Nit: undo the change please :)

**Review by joker-eph — APPROVED** [review]

File: mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir (hunk 2 above)
Comment (joker-eph) [reply mlir/test/Integration/Dialect/SparseTensor/GPU/CUDA/dump-ptx.mlir]: Yes I meant just checking that the flow is working, so even a SASS header or something is enough, thanks!
